anchore_engine/services/policy_engine/engine/vulns/providers.py

```diff
@@ -100,14 +106,26 @@ def load_image(self, image: Image, db_session, cache=False):
         # flush existing matches, recompute matches and add them to session
         scanner.flush_and_recompute_vulnerabilities(image, db_session=db_session)
 
+    def get_image_vulnerabilities_json(
+        self,
+        image: Image,
+        db_session,
+        vendor_only: bool = True,
+        force_refresh: bool = False,
+        cache: bool = True,
+    ) -> json:
+        return self.get_image_vulnerabilities(
+            image, db_session, vendor_only, force_refresh, cache
+        ).to_json()
```

Comment:
This seems outside the scope of the provider: formatting the output object.

Reply:
I had the jsonification outside of the provider scope initially but introduced it as an optimization for the grype-integration. When the report is loaded from cache, get_image_vulnerabilities_json() saves the additional steps to deserialize the json into object model and serialize it back again into json again. get_image_vulnerabilities() method will return the object model response, currently it's used by the policy evaluation gate to load vulnerabilities into the context